include/spdlog/sinks/win_eventlog_sink.h

```diff
@@ -0,0 +1,357 @@
+// Copyright(c) 2015-present, Gabi Melman & spdlog contributors.
+// Distributed under the MIT License (http://opensource.org/licenses/MIT)
+
+// Writing to Windows Event Log requires the registry entries below to be present, with the following modifications:
+// 1. {log_name}    should be replaced with your log name (e.g. your application name)
+// 2. {source_name} should be replaced with the specific source name and the key should be duplicated for
+//                  each source used in the application
+// 
+// Since typically modifications of this kind require elevation, it's better to do it as a part of setup procedure.
+// However, the win_eventlog_sink constructor can do it for you in runtime if you set the message_file_path parameter to 
+// win_eventlog_sink::DEFAULT_MESSAGE_FILE
+// 
+// You can also specify a custom message file if needed. 
+// Please refer to Event Log functions descriptions in MSDN for more details on custom message files.
+
+/*---------------------------------------------------------------------------------------
+
+Windows Registry Editor Version 5.00
+
+[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\EventLog\{log_name}]
+
+[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\EventLog\{log_name}\{source_name}]
+"TypesSupported"=dword:00000007
+"EventMessageFile"=hex(2):25,00,73,00,79,00,73,00,74,00,65,00,6d,00,72,00,6f,\
+  00,6f,00,74,00,25,00,5c,00,53,00,79,00,73,00,74,00,65,00,6d,00,33,00,32,00,\
+  5c,00,6d,00,73,00,63,00,6f,00,72,00,65,00,65,00,2e,00,64,00,6c,00,6c,00,00,\
+  00
+
+-----------------------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <winbase.h>
+
+#include <spdlog/details/null_mutex.h>
+#include <spdlog/sinks/base_sink.h>
+
+#include <mutex>
+#include <string>
+
+namespace spdlog {
+namespace sinks {
+
+namespace win_eventlog {
+
+namespace internal
+{
+
+struct utils
+{
+    /** Reports a message to stderr */
```

Comment:
not needed. just throw spdlog_ex

Reply:
This is used in two cases: one is when getting the SID for the current user fails for some reason, it still allows moving forward without the user id which is better than nothing.
Another thing is when something bad happens in the destructor, "just throw spdlog_ex" may end up with an exception during stack unwind and crash. So all the destructors just report any exceptions to stderr.

Reply:
suddenly printing to stderr might not be desired by users and even lead to bad performance. If the code need to move forward(like in the dtor), than just let it move forward, if not just throw..